testsuite/do_tests.sh.in

```diff
@@ -288,11 +289,18 @@ run_test ()
     sed 's/^/   > /g' "${TEST_OUTFILE}" >> "${TEST_LOGFILE}"
 
     msg_dirty=${param_success##* ${exit_code} }
+    msg_dirty_skip=${param_skipped##* ${exit_code} }
     msg_clean=${msg_dirty%%,*}
     if test "${msg_dirty}" != "${param_success}" ; then
         TEST_PASSED=$(( ${TEST_PASSED} + 1 ))
         explanation="${msg_clean}"
         junit_failure=
+    elif test "${msg_dirty_skip}" != "${param_skipped}" ; then
+        TEST_PASSED=$(( ${TEST_PASSED} + 1 ))
+        msg_dirty=${param_skipped##* ${exit_code} }
+        msg_clean=${msg_dirty%%,*}
+        explanation="${msg_clean}"
```

Comment:
If I understand this code right, then this section handles skipped tests. The Travis log shows that correct messages are shown for individual skipped tests, but the skipped tests should be counted as skipped, not as passed, and the final summary should show them as skipped.